src/scripts/main.js

```diff
@@ -1,3 +1,28 @@
 'use strict';
 
-// write code here
+const tree = document.querySelector('.tree');
+const listItems = document.querySelectorAll('li');
+
+listItems.forEach(item => {
+  if (item.firstElementChild) {
+    const span = document.createElement('span');
+
+    span.textContent = item.childNodes[0].textContent.trim();
+    item.childNodes[0].textContent = '';
+    item.prepend(span);
+  }
+});
+
+tree.addEventListener('click', e => {
+  if (e.target.tagName === 'SPAN') {
+    const children = e.target.parentNode.lastElementChild.children;
```

Comment:
It is better to check if 'e.target.parentNode.lastElementChild' is not null before accessing its children to avoid errors.